gqlalchemy/models.py

```diff
@@ -130,12 +131,71 @@ def __repr__(self) -> str:
         return str(self)
 
 
-class Node(UniqueGraphObject):
+class MyMeta(BaseModel.__class__):
+    def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
+        """This creates the class `Node`. It also creates all subclasses
+        of `Node`. Whenever a class is defined as a subclass of `Node`,
+        `MyMeta.__new__` is called.
+        """
+        cls = super().__new__(mcs, name, bases, namespace, **kwargs)
+        # TODO create a discussion about accessing labels through the class definition instead of through the object. E.g. `Person.labels` instead of `person = Person("Marko"); person._node_labels`.
+        cls._type = kwargs.get("_type", name)
+        cls._node_labels = kwargs.get("_node_labels", set(cls._type.split(":")))
+        labels = ":".join(cls._node_labels)
+        # TODO check if *_type* or *_node_labels* is in fields
```

Comment:
Why not check in this PR? What should happen in that case?

Reply:
Oh, I wanted to ask you about that. What if users don't want to create a sublcass of `Node` and they just want to use `Node` directly, this makes this much more complex (this is what that TODO would solve), I don't want to go into that on this launch.

This would enable the user to define special labels on which the index would be created, e.g.

```python
class Person(Node, _type="Person"):
    id: int = Field(index=True, _type="Person:Marko", db=Memgraph())
```

In this example, the index wouldn't be created on the label `Person` but rather on `Person:Marko`. I'm not sure how often this would be used.

Reply:
Let's move this under discussion, I wouldn't support this now.